pkg/controller/controller.go

```diff
@@ -866,11 +867,25 @@ func convertAndFilterCatalogToNamespacedTypes(namespace string, in *osb.CatalogR
 	return serviceClasses, servicePlans, nil
 }
 
+func generateEscapedName(externalID string) string {
+	re := regexp.MustCompile("[^a-zA-Z0-9]")
+
+	escapedName := re.ReplaceAllStringFunc(externalID, func(in string) string {
+		return fmt.Sprintf("-%x-", in)
```

Comment:
why do you have the trailing "-"?  If you use "-%0x" then you should always get a 2 digit number and don't need an end marker.

Reply:
I think this is a bit more complicated than that. The first number you provide is the default width, but as the value that is actually coming out is really a hex number, you need to provide a precision value to truncate it to two characters. I think you actualy want "-%0.1x", but I am not sure about that.

Reply:
Those don't work for actually weird characters like •

Reply:
ya, ID is globally-unique-string, which, assuming an character coding greater than ascii, means there are a lot of variable width stuff that can come out.

I don't get your output.

https://play.golang.org/p/v3mX7jtpgMG
returns
```
61  61  61
2009  09  09
```

for me, and if we expand it 
https://play.golang.org/p/Q76JiQw_HzH
```
package main

import "fmt"

func main() {
	printer("a")
	printer("\t")
	printer("t")
	printer("益")
	printer("🙈")
}

func printer(s string) {
	fmt.Printf("%0x  %2x  %0.1x\n", s, s, s) // tabs
}
```
yields
```
61  61  61
09  09  09
e79b8a  e79b8a  e7
f09f9988  f09f9988  f0
```
(I think tab maybe gets converted differently depending on where it's parsed?)

Reply:
I had an extra space before the first tab by mistake when I copied it into the comment - that's why you see the "20".

I think the `%2x` output seems fine since this is meant to be something people type of the cmd line and I sure hope they would avoid really odd chars like that when the rest of the line should be ascii.  However, I'm also fine with being really really worried and using @jberkhahn's original proposal of starting and ending with "-" to avoid possible conflicts.  Let's just pick one and go with it!